storage/StorageManager.cpp

```diff
@@ -96,23 +97,63 @@ DEFINE_int32(block_domain, 1,
 static const bool block_domain_dummy
     = gflags::RegisterFlagValidator(&FLAGS_block_domain, &ValidateBlockDomain);
 
-static bool ValidateBufferPoolSlots(const char *flagname,
-                                    std::uint64_t value) {
+/**
+ * @brief Set or validate the buffer pool slots. When automatically picking a
+ *        default value, check if the system is "small" or "large." Set the 
+ *        buffer pool space to 80% of the installed main memory for small
+ *        and 90% otherwise. 
+ *        This method follows the signature that is set by the gflags module.
+ * @param flagname The name of the buffer pool flag. 
+ * @param value The value of this flag from the command line, or default (0)
+ * @return True if the value was set to a legimate value, false otherwise. 
+ *         Currently this method aims to always find some legitimate value,
+ *         and never returns false.
+ **/
+static bool SetOrValidateBufferPoolSlots(const char *flagname,
+                                         std::uint64_t value) {
+  // Magic constants which may change in over time. The key idea is to leave
+  //   more memory (20%) on "small" systems for other processes.
+  const std::uint64_t a_giga_byte = (1<<30);
+  const std::uint64_t threshold_for_a_large_memory_system = 32;  // in GB.
+  const std::uint64_t percentage_to_grab_for_small_systems = 80;
+  const std::uint64_t percentage_to_grab_for_large_systems = 90;
```

Comment:
Will use const rather than constexpr as that is what is used in StorageConstants.hpp. We can have a separate PR to change all. But, in this case it won't matter.